src/PKSim.Core/Snapshots/Mappers/ClassificationSnapshotTask.cs

```diff
@@ -0,0 +1,88 @@
+{
+   public interface IClassificationSnapshotTask
+   {
+      /// <summary>
+      ///   Maps the <paramref name="snapshots"/> into classifications of type <paramref name="classificationType"/> and adds them to the <paramref name="project"/>.
+      ///   Also added to the project will be classifiables contained in the classifications and those will have the subjects
+      ///   configured correctly from the list of <paramref name="subjects"/>.
+      /// </summary>
+      /// <typeparam name="TClassifiable">This is the type of classifiable that is created for each classifiable in each newly created classification /></typeparam>
+      /// <typeparam name="TSubject">This is the type of the subject for the classifiable</typeparam>
+      Task UpdateProjectClassifications<TClassifiable, TSubject>(SnapshotClassification[] snapshots, PKSimProject project, IReadOnlyCollection<TSubject> subjects, ClassificationType classificationType) 
+         where TClassifiable : Classifiable<TSubject>, new() where TSubject : IObjectBase;
+
+      Task<SnapshotClassification[]> MapClassificationsToSnapshots(IReadOnlyList<ModelClassification> classifications, IReadOnlyList<IClassifiableWrapper> classifiables);
+   }
+
+   public class ClassificationSnapshotTask : IClassificationSnapshotTask
+   {
+      private readonly ClassificationMapper _classificationMapper;
+
+      public ClassificationSnapshotTask(ClassificationMapper classificationMapper)
+      {
+         _classificationMapper = classificationMapper;
+      }
+
+      public Task UpdateProjectClassifications<TClassifiable, TSubject>(SnapshotClassification[] snapshots, PKSimProject project, IReadOnlyCollection<TSubject> subjects, ClassificationType classificationType) 
+         where TClassifiable : Classifiable<TSubject>, new() where TSubject : IObjectBase
+      {
+         var tasks = snapshots.Select(snapshot => updateProjectFor<TClassifiable, TSubject>(project, snapshot, subjects, classificationType));
+
+         return Task.WhenAll(tasks);
+      }
+
+      private async Task updateProjectFor<TClassifiable, TSubject>(PKSimProject project, SnapshotClassification snapshot, IReadOnlyCollection<TSubject> subjects, ClassificationType classificationType, ModelClassification parent = null) 
+         where TClassifiable : Classifiable<TSubject>, new() where TSubject : IObjectBase
+      {
+         var model = await _classificationMapper.MapToModel(snapshot, classificationType);
+         model.Parent = parent;
+         project.AddClassification(model);
+
+         snapshot.Classifiables?.Each(snapshotClassifiable =>
+         {
+            var dataRepository = subjects.FirstOrDefault(x => string.Equals(x.Name, snapshotClassifiable));
+            var classifiable = project.GetOrCreateClassifiableFor<TClassifiable, TSubject>(dataRepository);
+            project.RemoveClassifiable(classifiable);
+            classifiable.Parent = model;
+            project.AddClassifiable(classifiable);
```

Comment:
The add and remove seems to be required to update the parent. I am not sure why this is like that internally. If this is required, we need a commment to explain why this is done like that

Reply:
I had seen this required at one stage before, but when I tested that today, it seems that only setting the parent is required. I removed the remove/add pairing